### ISSUE TITLE
Fast conversion between time formats

### ISSUE THREAD

**Issue by mhvk** [astropy/astropy#10596]
In https://github.com/astropy/astropy/issues/10587#issuecomment-661995131, @taldcroft wrote
> one thing I have pondered on and off is providing format_X to format_Y converter methods that bypass some of the Time machinery and basically go straight to ERFA.  I haven't gone any deeper than that level of thought, but if it were possible to make some very fast such methods I expect it would find use in the community. In other words one of main uses of `Time` is just as a conversion engine.

It might work as suggested in https://github.com/astropy/astropy/issues/10587#issuecomment-662049142
> something that would fall back on `Time(value, format=<from-format>).to_value(<to-format>)`, but has short-cuts for specific `from/to-format` combinations. Possibly, the fast converters could stored on the `to-format` `TimeFormat` classes?

**Comment by taldcroft** [comment]
@kangqiwang - interesting. Can you do the same testing for other formats like `isot`, `jd`?

**Comment by taldcroft** [comment]
@mhvk - I don't recall if I have mentioned it before, but I did a limited-scope implementation of this idea for the `CxoTime` subclass of `Time`. See:

- https://github.com/sot/cxotime/blob/master/cxotime/converters.py  (Current implementation with a static converters module)
- https://github.com/sot/cxotime/pull/33 (original PR, dynamically created converter functions)

This hand-picks a subset of formats that matter for Chandra operations and provides fast converters that bypass the `CxoTime` class initialization entirely. I wonder if it would be worth doing something like this for `Time`. There are issues trying to do this across the board for all formats, but it is certainly possible with a subset of common formats.

**Comment by mhvk** [comment]
Interesting! Looking at your converters, it looks like you always still go through `jd1` and `jd2`, so the old idea to possibly have some classmethods on the output formats might work (effectively, one has this already for the particular case of `jd1` and `jd2` in the right time scale, by using `from_jd=True`).

That said, I think it would make sense to try to we somehow get back the loss of speed in regular initialization first! It may also be an idea to see if we can implement some way to tell the `Time` initializer to by-pass verifications, format-finding, etc.

**Comment by taldcroft** [comment]
> Interesting! Looking at your converters, it looks like you always still go through `jd1` and `jd2`, so the old idea to possibly have some classmethods on the output formats might work (effectively, one has this already for the particular case of `jd1` and `jd2` in the right time scale, by using `from_jd=True`).

Right, this might be a good way to factor out the raw conversion code to/from JD.

> That said, I think it would make sense to try to we somehow get back the loss of speed in regular initialization first! It may also be an idea to see if we can implement some way to tell the `Time` initializer to by-pass verifications, format-finding, etc.

Yes, that's still something to consider. The risks there are:
- Might still be difficult to get good performance gains.
- Might end up making the code even more complex with a lot of conditional evaluations of code sections.

**Comment by kangqiwang** [comment]
## Astropy 7.0.1

```
%timeit at.Time(1, format='byear')
56.5 µs ± 1.54 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)

%timeit at.Time(1, format='cxcsec')
106 µs ± 21.2 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)

%timeit at.Time(1, format='gps')
105 µs ± 18.3 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)

%timeit at.Time("2000-01-01T00:00:00.000", format='isot')
101 µs ± 741 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)

%timeit at.Time(1, format='jd')
68.1 µs ± 13.8 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)

%timeit at.Time(1, format='jyear')
55.1 µs ± 719 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)

%timeit at.Time(1, format='mjd')
88 µs ± 17.8 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)

%timeit at.Time(1, format='plot_date')
272 µs ± 6.18 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

%timeit at.Time(1, format='unix')
93.2 µs ± 2.44 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)

%timeit at.Time(1, format='unix_tai')
96.4 µs ± 11.2 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)


```
